src/Init/Data/Array/Subarray.lean

```diff
@@ -9,25 +9,34 @@ import Init.Data.Array.Basic
 universe u v w
 
 structure Subarray (α : Type u)  where
-  as : Array α
+  array : Array α
   start : Nat
   stop : Nat
-  h₁ : start ≤ stop
-  h₂ : stop ≤ as.size
+  start_le_stop : start ≤ stop
+  stop_le_array_size : stop ≤ array.size
```

Comment:
```suggestion
  stop_le_size : stop ≤ array.size
```
this still seems sufficiently precise and is less annoyingly long

Reply:
Oh, yeah that is awkward. I wonder if we could change the name to avoid confusion (i.e. `Subarray.length`)? I think we will want names for both of these quantities eventually (e.g. in lemmas), so it would be good to think about what names are clear and distinct. `length` and `size`? `size` and `fullSize`? `extent` and `maxStop`? Or just use `t.array.size`.

As a pedantic note, prefix naming convention would yield `stop_le_size_array`.

Reply:
On reflection, I expect that this projection will not be used so often (mostly I expect users to be using the API here, or writing meaningful operators that preserve the invariants - borne out by the very small changes needed for Std and Aesop and the fact that Mathlib itself needed none), so I'll go with explicitness over concision.

Thanks!